src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,213 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail = new Node<>(tail, value, null);
+            tail.prev.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+        if ((index > size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't add element - index bigger than size");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        if (index == size - 1) {
+            addNode(tail.prev, value);
+            size++;
+            return;
+        }
+        Node<T> oldNode;
+        oldNode = getNode(index);
+        addNode(oldNode, value);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listElement: list) {
+            add(listElement);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't get element - index bigger than size");
+        }
+        if (index == 0) {
+            return head.element;
+        }
+        if (index == size - 1) {
+            return tail.element;
+        }
```

Comment:
do we need this checks? Could `getNode(index)` method just return head and tail in case of 0 and size - 1 indexes?